Contrib for an in-game reports system

InspectorCaracal opened the pull request:
#### Brief overview of PR changes/additions
This adds a contrib package for an in-game reporting system. The contrib is set up to add bug reports, player reports, and idea submissions, and provides a menu-based staff management command.

It uses global scripts and `Msg` objects under the hood, with very little custom logic, so it should work in just about anyone's game.

#### Motivation for adding to Evennia
It's a useful system that a lot of people wind up needing, so I thought I should share it!

InspectorCaracal commented:
Ah I messed something up, I'll fix it tomorrow.

Griatch commented on evennia/contrib/base_systems/ingame_reports/reports.py:
```diff
@@ -0,0 +1,284 @@
+        self.hub = hub
+        return super().at_pre_cmd()
+
+    def func(self):
```
Since this base class is intended to be overwritten, people may want to customize the `func' (like to search for something other than an Object). Not a blocker, but maybe break the parts up into smaller helpers to make it easier to override?

Griatch reviewed (changes requested):
This is a nice contrib. I also appreciate the work put into the README here. I have a few comments, some of which are more suggestions for clarification here and there.

InspectorCaracal commented on evennia/contrib/base_systems/ingame_reports/README.md:
```diff
@@ -0,0 +1,123 @@
+
+### Managing reports
+
+The `manage reports` command allows staff to review and manage the various types of reports. It dynamically aliases itself to include whatever types of reports you've configured for your game - by default, this means it makes `manage bugs`, `manage players`, and `manage ideas` available along with the default `manage reports`.
```
In case it changes your thought here: it isn't just `manage`, it's literally the string `manage reports`. The report type isn't an argument but a portion of dynamically-added command aliases.

Griatch replied:
Ah, I didn't notice that. That makes it less likely there'll be a clash, indeed. Very well.

Griatch commented:
@InspectorCaracal Ping :)

InspectorCaracal commented on evennia/contrib/base_systems/ingame_reports/README.md:
```diff
@@ -0,0 +1,123 @@
+
+### Managing reports
+
+The `manage reports` command allows staff to review and manage the various types of reports. It dynamically aliases itself to include whatever types of reports you've configured for your game - by default, this means it makes `manage bugs`, `manage players`, and `manage ideas` available along with the default `manage reports`.
```
Specifying that they're pluralized versions of the defined types is a good idea, but I think the rest of this was due to misinterpreting this as a poor explanation of `command <argument>` rather than actually being the commands themselves as is described. I'll figure out a way to be more explicit about that, instead.

InspectorCaracal commented:
I've updated the README to add or clarify the things that came up in here - especially for the command alias part, let me know if it's clear enough this time how it works.

I also split out the target search and the report creation steps into their own class methods to make them more overrideable, and moved the target/message argument parsing into `parse` for the same reason. Great suggestion, thank you!

Griatch commented on evennia/contrib/base_systems/ingame_reports/README.md:
```diff
@@ -0,0 +1,128 @@
+# In-Game Reporting System
+
```
See the [layout of a contrib](https://www.evennia.com/docs/latest/Contribs/Contribs-Guidelines.html#layout-of-a-contrib) and add `Contrib by Inspector Caracal 2024`.

Griatch reviewed (approved):
Thanks for the fixes. It's missing the contrib header credit, but I can add that when I merge.
